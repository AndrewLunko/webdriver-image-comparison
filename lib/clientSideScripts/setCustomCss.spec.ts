import {CssOptions} from './customCss.interfaces';
import setCustomCss from './setCustomCss';

describe('setCustomCss', () => {
  it('should be able to set the custom css with the default options', () => {
    const cssOptions:CssOptions = {
      addressBarPadding:6,
      disableCSSAnimation:false,
      id:'id',
      toolBarPadding:6,
    };

    expect(document.head.textContent).toMatchSnapshot();

    setCustomCss(cssOptions);

    expect(document.head.textContent).toMatchSnapshot();
  });

  it('should be able to set the custom css with the animations disabled', () => {
    const cssOptions:CssOptions = {
      addressBarPadding:6,
      disableCSSAnimation:true,
      id:'id',
      toolBarPadding:6,
    };

    expect(document.head.textContent).toMatchSnapshot();

    setCustomCss(cssOptions);

    expect(document.head.textContent).toMatchSnapshot();
  });

  it('should be able to set the custom css with the with padding set to 0', () => {
    const cssOptions:CssOptions = {
      addressBarPadding:0,
      disableCSSAnimation:true,
      id:'id',
      toolBarPadding:0,
    };

    expect(document.head.textContent).toMatchSnapshot();

    setCustomCss(cssOptions);

    expect(document.head.textContent).toMatchSnapshot();
  });
});
