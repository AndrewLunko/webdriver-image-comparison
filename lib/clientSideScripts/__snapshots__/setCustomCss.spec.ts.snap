// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`setCustomCss should be able to set the custom css with the animations disabled 1`] = `"body{padding-top:6px !important}body{padding-bottom:6px !important}"`;

exports[`setCustomCss should be able to set the custom css with the animations disabled 2`] = `
"body{padding-top:6px !important}body{padding-bottom:6px !important}
* {
    -o-transition-property: none !important;
    -moz-transition-property: none !important;
    -ms-transition-property: none !important;
    -webkit-transition-property: none !important;
    transition-property: none !important;
    -o-transform: none !important;
    -moz-transform: none !important;
    -ms-transform: none !important;
    -webkit-transform: none !important;
    transform: none !important;
    -webkit-animation: none !important;
    -moz-animation: none !important;
    -o-animation: none !important;
    -ms-animation: none !important;
    animation: none !important;
}body{padding-top:6px !important}body{padding-bottom:6px !important}"
`;

exports[`setCustomCss should be able to set the custom css with the default options 1`] = `""`;

exports[`setCustomCss should be able to set the custom css with the default options 2`] = `"body{padding-top:6px !important}body{padding-bottom:6px !important}"`;
